subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ConfigurationBuildDependenciesIntegrationTest.groovy

```diff
@@ -146,6 +150,9 @@ class ConfigurationBuildDependenciesIntegrationTest extends AbstractHttpDependen
 """
 
         when:
+        executer.expectDocumentedDeprecationWarning("""The resolved configuration 'compile' has been selected by the following variants:
```

Comment:
🤔 Trying to imagine how I'd interpret this error for the first time without the knowledge of what caused it: 

- In a multi-project build there may be many `compile` configurations.  I think the message should start with the project containing the bad configuration.  I know it appears later in the variants list.
- Instead of "selected" would it be clearer to say "consumed" or "would be consumed"?  Then the whole message aligns with the resolvable and consumable usage on the configuration that is the heart of the problem.
- Why is the variants list even relevant?  The resolved configuration is begin consumed and this is a problem.  Does it matter from where?  You have to make it not consumable if you're resolving it.  That's all the author needs to know to fix the issue, can't they dig into reporting if they want more detail?  I think this is a rare case of _too much_ info in the message.
- Instead of "role-based Configuration factory methods", which not suggest `resolvable()` directly?  That's the one they'll want to use.

Reply:
1. The name of the project is not accessible at this level. However, the user can switch the deprecations to a warning, and the project should be in the error message
2. Done
3. Removed
4. Done

Reply:
> has been consumed as a variant

The "as a variant" seems unnecessary to me.  How else would it be consumed?

> or use the 'resolvable(String)' configuration factory method. 

I'd append "to create them" to the end of that, otherwise people might be looking to call this method on their existing configuration.